### PR TITLE
Get heat map example working

### PR DESCRIPTION
**Pull request by graysonwright**
Source from http://bl.ocks.org/tjdecke/5558084

Decided not to complicate this with React.
React and d3 both have separate and conflicting ways to manage data in
documents. For the heatmap, d3's method will be easier to work with.

### CODE REVIEW

**Comment by paulcsmith** [comment]
I wonder if the code should be vendored for now if it's an example from somewhere else? Other than that I think this looks good

File: app/assets/javascripts/components/heatmap.es6.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+$(function() {
+  var margin = { top: 50, right: 0, bottom: 100, left: 30 };
+  var width = 960 - margin.left - margin.right;
+  var height = 430 - margin.top - margin.bottom;
+  var gridSize = Math.floor(width / 24);
+  var legendElementWidth = gridSize*2;
+  var buckets = 9;
+  var colors = ["#ffffd9","#edf8b1","#c7e9b4","#7fcdbb","#41b6c4","#1d91c0","#225ea8","#253494","#081d58"]; // alternatively colorbrewer.YlGnBu[9]
+  var days = ["Mo", "Tu", "We", "Th", "Fr", "Sa", "Su"];
+  var times = ["1a", "2a", "3a", "4a", "5a", "6a", "7a", "8a", "9a", "10a", "11a", "12a", "1p", "2p", "3p", "4p", "5p", "6p", "7p", "8p", "9p", "10p", "11p", "12p"];
```
Comment (graysonwright) [inline app/assets/javascripts/components/heatmap.es6.js]: These will soon be switched up when we change to the velocity/acceleration graph

File: app/assets/javascripts/components/heatmap.es6.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
+$(function() {
+  var margin = { top: 50, right: 0, bottom: 100, left: 30 };
+  var width = 960 - margin.left - margin.right;
+  var height = 430 - margin.top - margin.bottom;
+  var gridSize = Math.floor(width / 24);
+  var legendElementWidth = gridSize*2;
+  var buckets = 9;
+  var colors = ["#ffffd9","#edf8b1","#c7e9b4","#7fcdbb","#41b6c4","#1d91c0","#225ea8","#253494","#081d58"]; // alternatively colorbrewer.YlGnBu[9]
```
Comment (graysonwright) [inline app/assets/javascripts/components/heatmap.es6.js]: These will soon be switched up when we change to the velocity/acceleration graph

**Comment by graysonwright** [comment]
@paulcsmith I feel like we'll be editing the JS fairly soon. I'd rather keep it in `app` because it's more conventional. What's the motivation for `vendor`?

**Comment by paulcsmith** [comment]
I thought that it was code that was copy pasted from elsewhere and I wasn't sure we'd be using it or if it was just an example. If you're using it soon then I'm totally fine with merging it in as-is :)

**Pull request merged**